include/dsn/tool-api/network.h

```diff
@@ -328,6 +331,10 @@ class rpc_session : public ref_counter
     rpc_client_matcher *_matcher;
 
     std::atomic_int _delay_server_receive_ms;
+
+    // _client_username is only valid if it is a server rpc_session.
+    // it represents the name of the corresponding client
+    std::string _client_username;
```

Comment:
I think it's ok to add a `client_username` property for rpc_session. It's a stuff of the session itself, but not provided for security. security is just using it.

Reply:
No authorization/negotiation then no username. The RPC module should be pure network RPC implementation that can be used in every place whether or not the auth enabled. We must insist open-close principle that prevents such modification to add responsibility to a class.

I will take some time thinking about how to dispose of this variable, if you got no idea then.

Reply:
OK, but I don't suggest add user_name to negotiation, because in this way we should get the corresponding negotiation, which will acquire a lock. It will produce low efficiency